modules/module/BaseImageCaptionModel.py

```diff
@@ -62,9 +62,9 @@ def __is_supported_image_extension(filename: str) -> bool:
         if include_subdirectories:
             filenames = []
             for root, _, files in os.walk(sample_dir):
-                filenames.extend(os.path.join(root, fn) for fn in files if __is_supported_image_extension(fn))
+                filenames.extend(os.path.join(root, file) for file in files if __is_supported_image_extension(file))
```

Comment:
It's pedantic but could you use `name` since it's just the name. There's an established Python term for it and the contents of that variable is equivalent to this: https://docs.python.org/3/library/pathlib.html#pathlib.PurePath.name

Reply:
It would be nice to refactor this to use `Path.glob` somehow.